packages/app/src/cli/models/extensions/schemas.ts

```diff
@@ -32,17 +37,54 @@ export const ApiVersionSchema = zod.string()
 
 export type ApiVersionSchemaType = zod.infer<typeof ApiVersionSchema>
 
+export const SettingsSchema = zod.object({
+  fields: zod
+    .array(
+      zod.object({
+        key: zod.string().optional(),
+        name: zod.string().optional(),
+        description: zod.string().optional(),
+        required: zod.boolean().optional(),
+        type: zod.string(),
+      }),
+    )
+    .optional(),
+})
+
+export const HandleSchema = zod
+  .string()
+  .trim()
+  .min(3, 'Handle needs to have at least 3 characters')
+  .max(30, "Handle can't exceed 30 characters")
+  .regex(/^[a-zA-Z0-9.\-_]*$/, 'Handle can only contain alphanumeric characters, periods, hyphens, and underscores')
+
 export const BaseSchema = zod.object({
   name: zod.string(),
   type: zod.string(),
+  handle: HandleSchema.optional(),
   description: zod.string().optional(),
   api_version: ApiVersionSchema.optional(),
   extension_points: zod.any().optional(),
   capabilities: CapabilitiesSchema.optional(),
   metafields: zod.array(MetafieldSchema).optional().default([]),
   categories: zod.array(zod.string()).optional(),
+  extensions: zod.array(zod.any()).optional(),
```

Comment:
Would we ever allow a `shopify.extensions.toml` to not contain any `extensions`? For Flow extensions we set this property to be required and of `.min(1)`.

If this is set as optional to support legacy schemas would it make sense to add custom logic so that if the `extensions` array is there we require it to contain at least one extension?